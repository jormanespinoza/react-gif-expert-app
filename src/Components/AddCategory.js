import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => setInputValue(e.target.value);

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(categories => [inputValue, ...categories]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <span className="hidden">{inputValue}</span>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
