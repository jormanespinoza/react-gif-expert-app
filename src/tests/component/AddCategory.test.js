import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../Components/AddCategory';

describe('Tests on AddCategory component', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('should render <AddCategory /> properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change on input change', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo'
        input.simulate('change', { target: { value } });

        expect(wrapper.find('.hidden').text().trim()).toBe(value);
    });

    test('should post anything on submit when given input is not valid', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call setCategories and empty inputValue', () => {
        const value = 'Hola Mundo'
        const input = wrapper.find('input');
        input.simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault() {} });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(input.prop('value')).toBe('')
    });
});
