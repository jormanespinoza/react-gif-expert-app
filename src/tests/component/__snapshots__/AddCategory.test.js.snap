// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on AddCategory component should render <AddCategory /> properly 1`] = `
<form
  onSubmit={[Function]}
>
  <span
    className="hidden"
  />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
